src/net/discovery/mod.rs

```diff
@@ -40,5 +41,5 @@ pub mod peer {
 /// Any potential failures should be captured in the initialization of the implementer.
 #[allow(clippy::module_name_repetitions)] // following standard naming convention
 pub trait PeerDiscovery {
-    fn peers(&self) -> &[peer::Config; 3];
+    fn peers(&self) -> [peer::Config; 3];
```

Comment:
I think we should keep the previous interface. You can clone at the client side if you need to but after our last conversation I thought we don't need t hat

Reply:
I thought it would make more sense to switch to owned so that way, the `HttpHelper` can own the peer config, and pass it out as reference to queries when they need them. If I switch this back to a reference, then the `HttpHelper` must hold the `PeerDiscovery` itself, which means it needs to carry a generic parameter with it. I thought this was a cleaner approach while still having no copies.

What is the advantage to holding the `PeerDiscovery` vs the `peer:Config`?

Reply:
> aren't you making a fresh copy when peers is called?

yes, but `peers()` is called only once, by `HttpHelper`. it is passed around by reference from there. I don't think there'd be anything that would need access to `peer_discovery` that wouldn't have access to `HttpHelper`.

> Components lazily grab the whole config only to pick a couple of configuration options later.

would a better model be to divide up the config into areas of interest? so you'd have a config divided up into "http" vs "prss" vs anything else?

as of now, it's divided up into "h1", "h2", "h3", but that can be changed

Reply:
> yes, but peers() is called only once, by HttpHelper. 

My worry is that this design is based on assumptions that may no longer be true in the future. I would much rather prefer a cleaner API that allows inefficient actors (cloning at the callsite) if we really need to. Maybe I misunderstand something but I don't think we even need that here. 

> would a better model be to divide up the config into areas of interest? so you'd have a config divided up into "http" vs "prss" vs anything else?

To some extent it would help and I believe you already have it in the config (I see `http` and `prss` sections there). There might be some hiccups later when you may want to use config value across multiple layers and components, but otherwise it should work. The key part imo that components should only ask for things they need and don't get anything extra. Note that there is a way to achieve that with no extra structs being created by using traits,

Reply:
ok, I get your point about this now. I will modify.

I can also separate the config into different concerns if that would be more clear. don't think we'll need any traits for now, but possibly in the future.

That being said, I will separate the config within the other branch https://github.com/thurstonsand/raw-ipa/pull/8 because that branch actually deletes all of the prss config stuff since it is no longer necessary

Reply:
addressed.

you can see the separation of concerns change here: https://github.com/thurstonsand/raw-ipa/pull/8/commits/4e800f7ec5134bb703ee77da25343ee2410f39ec